### PR TITLE
Support for referring to other globals from globals

### LINKED ISSUE #175: Global cannot access other globals
Testcase: 

```qml
global Foo := {
    property <string> hi: "Hi";
}

global Bar := {
    property <string> ho: Foo.hi;
}

export Demo := Window {
    Text {
        text: Bar.ho;
    }
}
```

Tested at least with the interpreter which panics.

### PR DESCRIPTION
**Pull request by ogoffart**
- We need to make sure that the initialization of global is in the right order.
 - In C++ and rust, we need to add accessor to the global component
 - There can be `PropertyReference::Global` in binding of globals
 - The interpreter globals need to hold references to the global they may depend on
    
Fixes #175

### CODE REVIEW

File: internal/interpreter/global_component.rs [hunk 1]
```diff
@@ -107,28 +110,27 @@ pub fn instantiate(description: &CompiledGlobal) -> (String, Pin<Rc<dyn GlobalCo
...
                 None,
                 None,
-            )));
-            let id = if component.is_global() {
-                component.root_element.borrow().id.clone()
-            } else {
-                component.id.clone()
-            };
-            (id, g)
+                globals.clone(),
```
Comment (tronical) [inline internal/interpreter/global_component.rs:124]: This clones the entire HashMap. Maybe it's worth putting that into an RC?

**Review by tronical — APPROVED** [review]
Nice!

File: internal/interpreter/global_component.rs (hunk 1 above)
Comment (ogoffart) [reply internal/interpreter/global_component.rs:124]: a `Rc<RefCell<HashMap<String, Rc<_>>`
Well no that won't work. there would be cycle.

I thought about trying to take the Hashmap and then take it back, but then i thought there is no point. The hashmap is containing Rc anyway.

Yes, that means this use O(n²) memory in the number of globals.  But there should not be that many globals anyway.

Comment (tronical) [reply internal/interpreter/global_component.rs:124]: True true, there shouldn't be many and this design is much simpler.

**Pull request merged**